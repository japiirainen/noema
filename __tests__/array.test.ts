import { array, number, object } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

describe("array", () => {
  test("should parse a valid array", () => {
    expectParseSuccess(array(number), [1])
  })

  test("should fail to parse non-arrays", () => {
    expectParseFailure(array(number), 0)
  })

  test("should parse empty arrays", () => {
    expectParseSuccess(array(number), [])
  })

  test("should fail to parse arrays with invalid values", () => {
    expectParseFailure(array(number), [1, "2", 3])
  })

  test("the path property works within an object", () => {
    expectParseFailure(object({ a: array(number) }), { a: ["1"] })
  })
})
