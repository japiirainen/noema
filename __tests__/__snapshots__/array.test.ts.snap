// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`array should fail to parse arrays with invalid values 1`] = `
[
  {
    "code": "invalid_number",
    "message": "Expected value to be a number",
    "path": "1",
    "value": "2",
  },
]
`;

exports[`array should fail to parse non-arrays 1`] = `
[
  {
    "code": "invalid_array",
    "message": "Expected value to be an array",
    "path": "",
    "value": 0,
  },
]
`;

exports[`array the path property works within an object 1`] = `
[
  {
    "code": "invalid_number",
    "message": "Expected value to be a number",
    "path": "a[0]",
    "value": "1",
  },
]
`;
